p_array

I tried to clone, then build this package.

I get the following error on `Pkg.build("Xtensor")`

```
In file included from /Users/sglyon/.julia/v0.6/Xtensor/deps/xtensor-julia-examples/tensors.cpp:3:
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:277:27: error: no member named 'p_array'
      in 'xt::jltensor<double, 2>'
        std::copy(&(this->p_array->nrows), &(this->p_array->nrows) + N, m_shape.begin());
                    ~~~~  ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:279:65: error: no member named 'p_array'
      in 'xt::jltensor<double, 2>'
        m_data = container_type(reinterpret_cast<pointer>(this->p_array->data),
```

Not sure why this is happening as `p_array` does seem to be a member of the parent class `jlcontainer`.

Here's the whole report (probably more than you wanted, but putting here just in case):


```
julia> Pkg.build("Xtensor")

WARNING: deprecated syntax "_ as an rvalue".
INFO: Building CxxWrap
Found Julia library at /Users/sglyon/src/julia/usr/lib/libjulia.dylib
INFO: Changing Directory to /Users/sglyon/.julia/v0.6/CxxWrap/deps/builds/cxx_wrap
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/sglyon/.julia/v0.6/CxxWrap/deps/builds/cxx_wrap
[100%] Built target cxx_wrap
Install the project...
-- Install configuration: "Release"
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libcxx_wrap.0.1.dylib
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libcxx_wrap.0.dylib
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libcxx_wrap.dylib
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: no LC_RPATH load command with path: /Users/sglyon/src/julia/usr/lib found in: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libcxx_wrap.0.1.dylib (for architecture x86_64), required for specified option "-delete_rpath /Users/sglyon/src/julia/usr/lib"
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/array.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/functions.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/smart_pointers.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/type_conversion.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/containers/const_array.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/containers/tuple.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/cmake/CxxWrapConfig.cmake
-- Installing: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/cmake/CxxWrapConfig-release.cmake
-- Up-to-date: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/cmake/CxxWrapConfigVersion.cmake
INFO: Changing Directory to /Users/sglyon/.julia/v0.6/CxxWrap/deps/builds/examples
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/sglyon/.julia/v0.6/CxxWrap/deps/builds/examples
[ 12%] Built target functions
[ 87%] Built target except
[100%] Built target hello
[100%] Built target extended
[100%] Built target cxxwrap_containers
[100%] Built target parametric
[100%] Built target types
[100%] Built target inheritance
Install the project...
-- Install configuration: "Release"
-- Installing: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libcxxwrap_containers.dylib
-- Installing: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libexcept.dylib
-- Installing: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libextended.dylib
-- Installing: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libfunctions.dylib
-- Installing: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libhello.dylib
-- Installing: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libinheritance.dylib
-- Installing: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libparametric.dylib
-- Installing: /Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/lib/libtypes.dylib
INFO: Building Xtensor
Building xtensor-core
Building xtensor-julia
Building xtensorjl
INFO: Attempting to Create directory /Users/sglyon/.julia/v0.6/Xtensor/builds/xtensor
INFO: Directory /Users/sglyon/.julia/v0.6/Xtensor/builds/xtensor already created
INFO: Changing Directory to /Users/sglyon/.julia/v0.6/Xtensor/builds/xtensor
Cloning into '/Users/sglyon/.julia/v0.6/Xtensor/deos/xtensor'...
remote: Counting objects: 4240, done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 4240 (delta 4), reused 22 (delta 2), pack-reused 4214
Receiving objects: 100% (4240/4240), 1.47 MiB | 2.34 MiB/s, done.
Resolving deltas: 100% (2923/2923), done.
Note: checking out '2b3c248538b8647aaed04e10484bb35f91157a82'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

-- xtensor v0.9.0
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/sglyon/.julia/v0.6/Xtensor/builds/xtensor
Install the project...
-- Install configuration: ""
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xarray.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xassign.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xbroadcast.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xbuilder.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xcontainer.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xeval.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xexception.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xexpression.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xfunction.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xgenerator.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xindexview.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xio.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xiterable.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xiterator.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xlayout.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xmath.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xnoalias.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xoperation.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xrandom.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xreducer.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xscalar.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xsemantic.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xstridedview.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xslice.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xstorage.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xstrides.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xtensor.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xtensor_config.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xtensor_forward.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xutils.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xvectorize.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xview.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xoptional.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xmissing.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xfunctorview.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xoffsetview.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xcomplex.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xview_utils.hpp
-- Installing: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor/xcsv.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/share/cmake/xtensor/xtensorConfig.cmake
-- Up-to-date: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/share/cmake/xtensor/xtensorConfigVersion.cmake
INFO: Attempting to Create directory /Users/sglyon/.julia/v0.6/Xtensor/builds/xtensor-julia
INFO: Directory /Users/sglyon/.julia/v0.6/Xtensor/builds/xtensor-julia already created
INFO: Changing Directory to /Users/sglyon/.julia/v0.6/Xtensor/builds/xtensor-julia
-- xtensor-julia v..
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/sglyon/.julia/v0.6/Xtensor/builds/xtensor-julia
Install the project...
-- Install configuration: ""
-- Up-to-date: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jlbuffer_adaptor.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jlcontainer.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/xtensor_julia_config.hpp
-- Up-to-date: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/share/cmake/xtensor-julia/xtensor-juliaConfig.cmake
-- Up-to-date: /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/share/cmake/xtensor-julia/xtensor-juliaConfigVersion.cmake
INFO: Attempting to Create directory /Users/sglyon/.julia/v0.6/Xtensor/deps/builds/xtensor-julia-examples
INFO: Directory /Users/sglyon/.julia/v0.6/Xtensor/deps/builds/xtensor-julia-examples already created
INFO: Changing Directory to /Users/sglyon/.julia/v0.6/Xtensor/deps/builds/xtensor-julia-examples
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/sglyon/.julia/v0.6/Xtensor/deps/builds/xtensor-julia-examples
[ 50%] Building CXX object CMakeFiles/tensors.dir/tensors.cpp.o
In file included from /Users/sglyon/.julia/v0.6/Xtensor/deps/xtensor-julia-examples/tensors.cpp:3:
In file included from /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:22:
In file included from /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jlcontainer.hpp:17:
In file included from /Users/sglyon/anaconda3/include/xtensor/xcontainer.hpp:19:
In file included from /Users/sglyon/anaconda3/include/xtensor/xmath.hpp:21:
In file included from /Users/sglyon/anaconda3/include/xtensor/xreducer.hpp:28:
/Users/sglyon/anaconda3/include/xtensor/xbuilder.hpp:714:23: warning: comparison of unsigned expression < 0 is always
      false [-Wtautological-compare]
        if (diag_size < 0)
            ~~~~~~~~~ ^ ~
In file included from /Users/sglyon/.julia/v0.6/Xtensor/deps/xtensor-julia-examples/tensors.cpp:3:
In file included from /Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:22:
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jlcontainer.hpp:61:26: error: unknown type name
      'layout_type'
        static constexpr layout_type static_layout = layout_type::column_major;
                         ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jlcontainer.hpp:61:54: error: use of undeclared
      identifier 'layout_type'
        static constexpr layout_type static_layout = layout_type::column_major;
                                                     ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jlcontainer.hpp:66:9: error: unknown type name
      'layout_type'
        layout_type layout() const;
        ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jlcontainer.hpp:132:51: error: unknown type name
      'layout_type'
    inline auto jlcontainer<D>::layout() const -> layout_type
                                                  ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jlcontainer.hpp:134:16: error: use of undeclared
      identifier 'layout_type'
        return layout_type::column_major;
               ^
In file included from /Users/sglyon/.julia/v0.6/Xtensor/deps/xtensor-julia-examples/tensors.cpp:3:
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:134:34: error: use of undeclared
      identifier 'layout_type'
        compute_strides(m_shape, layout_type::column_major, m_strides, m_backstrides);
                                 ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:159:34: error: use of undeclared
      identifier 'layout_type'
        compute_strides(m_shape, layout_type::column_major, m_strides, m_backstrides);
                                 ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:174:34: error: use of undeclared
      identifier 'layout_type'
        compute_strides(m_shape, layout_type::column_major, m_strides, m_backstrides);
                                 ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:201:34: error: use of undeclared
      identifier 'layout_type'
        compute_strides(m_shape, layout_type::column_major, m_strides, m_backstrides);
                                 ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:231:34: error: use of undeclared
      identifier 'layout_type'
        compute_strides(m_shape, layout_type::column_major, m_strides, m_backstrides);
                                 ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:278:34: error: use of undeclared
      identifier 'layout_type'
        compute_strides(m_shape, layout_type::column_major, m_strides, m_backstrides);
                                 ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:73:60: error: no type named
      'inner_backstrides_type' in 'xt::jlcontainer<xt::jltensor<double, 2> >'
        using inner_backstrides_type = typename base_type::inner_backstrides_type;
                                       ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/Users/sglyon/.julia/v0.6/Xtensor/deps/xtensor-julia-examples/tensors.cpp:7:54: note: in instantiation of template
      class 'xt::jltensor<double, 2>' requested here
    double test_access(const xt::jltensor<double, 2> m)
                                                     ^
In file included from /Users/sglyon/.julia/v0.6/Xtensor/deps/xtensor-julia-examples/tensors.cpp:3:
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:92:15: error: using declaration refers
      into 'base_type::', which is not a base class of 'jltensor<double, 2>'
        using base_type::begin;
              ^~~~~~~~~~~
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:93:15: error: using declaration refers
      into 'base_type::', which is not a base class of 'jltensor<double, 2>'
        using base_type::end;
              ^~~~~~~~~~~
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:184:11: error: type 'base_type' (aka
      'jlcontainer<jltensor<double, 2UL> >') is not a direct or virtual base of 'xt::jltensor<double, 2>'
        : base_type(jl)
          ^~~~~~~~~
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:353:20: note: in instantiation of member
      function 'xt::jltensor<double, 2>::jltensor' requested here
            return xt::jltensor<T, N>(arr);
                   ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/containers/../type_conversion.hpp:248:10: note: in instantiation of
      member function 'cxx_wrap::ConvertToCpp<xt::jltensor<double, 2>, false, false, false, void>::operator()'
      requested here
  return cpp_converter_type<CppT>()(julia_val);
         ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:33:41: note: in instantiation of function template
      specialization 'cxx_wrap::convert_to_cpp<xt::jltensor<double, 2>, jl_array_t *>' requested here
    return convert_to_julia((*std_func)(convert_to_cpp<mapped_reference_type<Args>>(args)...));
                                        ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:56:14: note: in instantiation of member function
      'cxx_wrap::detail::ReturnTypeAdapter<double, xt::jltensor<double, 2> >::operator()' requested here
      return ReturnTypeAdapter<R, Args...>()(functor, args...);
             ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:187:69: note: in instantiation of member function
      'cxx_wrap::detail::CallFunctor<double, xt::jltensor<double, 2> >::apply' requested here
    return reinterpret_cast<void*>(detail::CallFunctor<R, Args...>::apply);
                                                                    ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:181:3: note: in instantiation of member function
      'cxx_wrap::FunctionWrapper<double, xt::jltensor<double, 2> >::pointer' requested here
  FunctionWrapper(const functor_t& function) : FunctionWrapperBase(julia_return_type<R>()), m_function(function)
  ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:374:29: note: in instantiation of member function
      'cxx_wrap::FunctionWrapper<double, xt::jltensor<double, 2> >::FunctionWrapper' requested here
    auto* new_wrapper = new FunctionWrapper<R, Args...>(f);
                            ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:389:14: note: in instantiation of function template
      specialization 'cxx_wrap::Module::method<double, xt::jltensor<double, 2> >' requested here
      return method(name, std::function<R(Args...)>(f));
             ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/xtensor-julia-examples/tensors.cpp:15:13: note: in instantiation of function
      template specialization 'cxx_wrap::Module::method<double, xt::jltensor<double, 2> >' requested here
        mod.method("test_access", test_access);
            ^
In file included from /Users/sglyon/.julia/v0.6/Xtensor/deps/xtensor-julia-examples/tensors.cpp:3:
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:272:19: error: no member named 'p_array'
      in 'xt::jltensor<double, 2>'
        if (this->p_array->flags.ndims != N)
            ~~~~  ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:186:9: note: in instantiation of member
      function 'xt::jltensor<double, 2>::init_from_julia' requested here
        init_from_julia();
        ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:353:20: note: in instantiation of member
      function 'xt::jltensor<double, 2>::jltensor' requested here
            return xt::jltensor<T, N>(arr);
                   ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/containers/../type_conversion.hpp:248:10: note: in instantiation of
      member function 'cxx_wrap::ConvertToCpp<xt::jltensor<double, 2>, false, false, false, void>::operator()'
      requested here
  return cpp_converter_type<CppT>()(julia_val);
         ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:33:41: note: in instantiation of function template
      specialization 'cxx_wrap::convert_to_cpp<xt::jltensor<double, 2>, jl_array_t *>' requested here
    return convert_to_julia((*std_func)(convert_to_cpp<mapped_reference_type<Args>>(args)...));
                                        ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:56:14: note: in instantiation of member function
      'cxx_wrap::detail::ReturnTypeAdapter<double, xt::jltensor<double, 2> >::operator()' requested here
      return ReturnTypeAdapter<R, Args...>()(functor, args...);
             ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:187:69: note: in instantiation of member function
      'cxx_wrap::detail::CallFunctor<double, xt::jltensor<double, 2> >::apply' requested here
    return reinterpret_cast<void*>(detail::CallFunctor<R, Args...>::apply);
                                                                    ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:181:3: note: in instantiation of member function
      'cxx_wrap::FunctionWrapper<double, xt::jltensor<double, 2> >::pointer' requested here
  FunctionWrapper(const functor_t& function) : FunctionWrapperBase(julia_return_type<R>()), m_function(function)
  ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:374:29: note: in instantiation of member function
      'cxx_wrap::FunctionWrapper<double, xt::jltensor<double, 2> >::FunctionWrapper' requested here
    auto* new_wrapper = new FunctionWrapper<R, Args...>(f);
                            ^
/Users/sglyon/.julia/v0.6/CxxWrap/deps/usr/include/cxx_wrap.hpp:389:14: note: in instantiation of function template
      specialization 'cxx_wrap::Module::method<double, xt::jltensor<double, 2> >' requested here
      return method(name, std::function<R(Args...)>(f));
             ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/xtensor-julia-examples/tensors.cpp:15:13: note: in instantiation of function
      template specialization 'cxx_wrap::Module::method<double, xt::jltensor<double, 2> >' requested here
        mod.method("test_access", test_access);
            ^
In file included from /Users/sglyon/.julia/v0.6/Xtensor/deps/xtensor-julia-examples/tensors.cpp:3:
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:277:27: error: no member named 'p_array'
      in 'xt::jltensor<double, 2>'
        std::copy(&(this->p_array->nrows), &(this->p_array->nrows) + N, m_shape.begin());
                    ~~~~  ^
/Users/sglyon/.julia/v0.6/Xtensor/deps/usr/include/xtensor-julia/jltensor.hpp:279:65: error: no member named 'p_array'
      in 'xt::jltensor<double, 2>'
        m_data = container_type(reinterpret_cast<pointer>(this->p_array->data),
                                                          ~~~~  ^
1 warning and 18 errors generated.
make[2]: *** [CMakeFiles/tensors.dir/tensors.cpp.o] Error 1
make[1]: *** [CMakeFiles/tensors.dir/all] Error 2
make: *** [all] Error 2
==================================================[ ERROR: Xtensor ]==================================================

LoadError: failed process: Process(`cmake --build . --config Release --target install -- -j 10`, ProcessExited(2)) [2]
while loading /Users/sglyon/.julia/v0.6/Xtensor/deps/build.jl, in expression starting on line 84

======================================================================================================================

===================================================[ BUILD ERRORS ]===================================================

WARNING: Xtensor had build errors.

 - packages with build errors remain installed in /Users/sglyon/.julia/v0.6
 - build the package(s) and all dependencies with `Pkg.build("Xtensor")`
 - build a single package by running its `deps/build.jl` script

======================================================================================================================

shell> gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

shell> clang --version
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

shell> cmake --version
cmake version 3.8.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).

shell> make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
```

--- a/deps/build.jl
+++ b/deps/build.jl
@@ -36,7 +36,7 @@ xtensor_core_steps = @build_steps begin
 end
 
 xtensor_julia_steps = @build_steps begin
-  `cmake -G "$genopt" -DCMAKE_INSTALL_PREFIX="$prefix" $xtensor_julia_srcdir`
+  `cmake -G "$genopt" -DCMAKE_PREFIX_PATH=$prefix -DCMAKE_INSTALL_PREFIX=$prefix -Dxtensor_DIR=$xtensor_dir $xtensor_julia_srcdir`
   `cmake --build . --config $build_type --target install $makeopts`
 end
 
@@ -52,7 +52,7 @@ for l in example_labels
 end
 
 xtensorjl_steps = @build_steps begin
-  `cmake -G "$genopt" -DCMAKE_INSTALL_PREFIX="$prefix" -DCMAKE_BUILD_TYPE="$build_type" -DLIBDIR_SUFFIX=$libdir_opt -DCxxWrap_DIR=$cxx_wrap_dir -Dxtensor_DIR=$xtensor_dir $xtensorjl_srcdir`
+  `cmake -G "$genopt" -DCMAKE_PREFIX_PATH=$prefix -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_BUILD_TYPE="$build_type" -DLIBDIR_SUFFIX=$libdir_opt -DCxxWrap_DIR=$cxx_wrap_dir -Dxtensor_DIR=$xtensor_dir $xtensorjl_srcdir`
   `cmake --build . --config $build_type --target install $makeopts`
 end
 

